sn_networking/src/transfers.rs

```diff
@@ -156,64 +154,15 @@ impl Network {
         let mut our_output_cash_notes = Vec::new();
 
         for (id, derivation_index) in our_output_unique_pubkeys.into_iter() {
-            let src_tx = parent_txs
-                .iter()
-                .find(|tx| tx.outputs.iter().any(|o| o.unique_pubkey() == &id))
-                .ok_or(NetworkError::InvalidTransfer(
-                    "None of the CashNoteRedemptions are destined to our key".to_string(),
-                ))?
-                .clone();
-            let signed_spends: BTreeSet<SignedSpend> = parent_spends
-                .iter()
-                .filter(|s| s.spent_tx_hash() == src_tx.hash())
-                .cloned()
-                .collect();
             let cash_note = CashNote {
                 unique_pubkey: id,
-                parent_tx: src_tx,
-                parent_spends: signed_spends,
+                parent_spends: parent_spends.clone(),
                 main_pubkey,
                 derivation_index,
             };
             our_output_cash_notes.push(cash_note);
         }
 
-        // check Txs and parent spends are valid
-        trace!("Validating parent spends");
-        for tx in parent_txs {
-            let tx_inputs_keys: Vec<_> = tx.inputs.iter().map(|i| i.unique_pubkey()).collect();
-
-            // get the missing inputs spends from the network
-            let mut tasks = JoinSet::new();
-            for input_key in tx_inputs_keys {
-                if parent_spends.iter().any(|s| s.unique_pubkey() == input_key) {
-                    continue;
-                }
-                let self_clone = self.clone();
-                let addr = SpendAddress::from_unique_pubkey(input_key);
-                let _ = tasks.spawn(async move { self_clone.get_spend(addr).await });
-            }
-            while let Some(result) = tasks.join_next().await {
-                let signed_spend = result
-                    .map_err(|e| NetworkError::FailedToGetSpend(format!("{e}")))?
-                    .map_err(|e| NetworkError::InvalidTransfer(format!("{e}")))?;
-                let _ = parent_spends.insert(signed_spend.clone());
-            }
-
-            // verify the Tx against the inputs spends
-            let input_spends: BTreeSet<_> = parent_spends
-                .iter()
-                .filter(|s| s.spent_tx_hash() == tx.hash())
-                .cloned()
-                .collect();
-            tx.verify_against_inputs_spent(&input_spends).map_err(|e| {
-                NetworkError::InvalidTransfer(format!(
-                    "Payment parent Tx {:?} invalid: {e}",
-                    tx.hash()
-                ))
-            })?;
-        }
```

Comment:
Here we've removed all sort of transfer verification.
Letting people accepting transfers of money that isn't valid!

Reply:
If this passed CI checks, is there a test we can add for this so it's not missed in future?

Reply:
Yes we should add a test for this!